pywbem_mock/_instancewriteprovider.py

```diff
@@ -270,367 +209,342 @@ def __init__(self, cimrepository=None):
-                if p not in property_list:
-                    del modified_instance[p]
-
-        # Exception if property in instance but not class or types do not
-        # match
-        for pname in modified_instance:
-            if pname not in target_class.properties:
-                raise CIMError(
-                    CIM_ERR_INVALID_PARAMETER,
-                    _format("Property {0!A} specified in ModifiedInstance is "
-                            "not exposed by class {1!A} in namespace {2!A}",
-                            pname, target_class.classname, namespace))
-
-            classprop = target_class.properties[pname]
-            instprop = modified_instance.properties[pname]
-            if instprop.is_array != classprop.is_array \
-                    or instprop.type != classprop.type \
-                    or instprop.array_size != classprop.array_size:
-                raise CIMError(
-                    CIM_ERR_INVALID_PARAMETER,
-                    _format("Instance and class property name={0!A} type "
-                            "or other attributes do not match: "
-                            "instance={1!A}, class={2!A}",
-                            pname, instprop, classprop))
-            # If case of modified_instance property != case of class property
-            # change the name in the modified_instance
-            if instprop.name != classprop.name:
-                modified_instance.properties[instprop.name].name = \
-                    classprop.name
-
-        # Modify the value of properties in the repo with those from
-        # modified instance inserted into the original instance
-        orig_instance.update(modified_instance.properties)
-        instance_store.update(modified_instance.path, orig_instance)
-
-        return
+        instance = instance_store.get(ModifiedInstance.path, copy=False)
+
+        # Set ModifiedInstance to have just the properties to be modified
+        if PropertyList is not None:
+
+            # Create property set with only unique names from PropertyList
+            # The property names are stored in lower case to support case
+            # insensitivity properly.
+            property_set = set()
+            for pn in PropertyList:
+                property_set.add(pn.lower())
+
+            # Add class default values for properties not specified in
+            # ModifiedInstance.
+            for pn in property_set:
+                if pn not in ModifiedInstance:
+                    # If the property in the class does not have a default
+                    # value, it is None.
+                    ModifiedInstance[pn] = creation_class.properties[pn].value
+
+            # Remove properties from ModifiedInstance that are not in
+            # PropertyList.
+            for pn in list(ModifiedInstance):
+                if pn.lower() not in property_set:
+                    del ModifiedInstance[pn]
```

Comment:
This is a general rule in DSP0200, and could thus be moved up to the dispatcher.

DISCUSSION

Reply:
We agreed to go for that.

Reply:
DONE in latest commit.